modules/common/src/main/scala/surge/kafka/streams/AggregateStateStoreKafkaStreamsImpl.scala

```diff
@@ -55,6 +56,24 @@ private[streams] class AggregateStateStoreKafkaStreamsImpl[Agg >: Null](
     StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG -> classOf[AggregateStreamsRocksDBConfig].getName,
     StreamsConfig.APPLICATION_SERVER_CONFIG -> settings.localActorHostPort)
 
+  private val getSubStateForAggregateTimerMetric: Timer = metrics.timer(
+    MetricInfo(
+      name = s"surge.${aggregateName.toLowerCase()}.get-subState-aggregate",
+      description = "The time taken to get the sub-state of the aggregate",
+      tags = Map("aggregate" -> aggregateName)))
+
+  private val getAggregateBytesTimerMetric: Timer = metrics.timer(
+    MetricInfo(
+      name = s"surge.${aggregateName.toLowerCase()}.get-aggregate-state",
+      description = "The time taken to getting the aggregate state.",
+      tags = Map("aggregate" -> aggregateName)))
+
+  val current: Rate = metrics.rate(
+    MetricInfo(
```

Comment:
I think we've already got this metric on the producer actor somewhere - this is capturing rates on the KTable read side but I'd be curious if we can capture metrics on the KTable write side. It may be that Kafka Streams already has some metrics for that, but I haven't looked too closely. Unfortunately in this case (but good for other reasons) our stream topology is so simple that there's nowhere really to put a custom metric for rate on the input side

Reply:
Yeah sure i will look for the same meanwhile